From 297990324346921550e1527b3b4b468e6e18aa47 Mon Sep 17 00:00:00 2001
From: Damian Eppel <deppel@marvell.com>
Date: Wed, 26 Jan 2022 01:54:52 -0800
Subject: [PATCH 11/12] phy: mdio-thunder: stop toggling SMI clock on idle

commit 5541a026c6987c638dcb95a8da1216e87679d23f from
git@git.assembla.com:cavium/WindRiver.linux.git

SMI clock should be running only for the time when
there are transactions on the bus.

Signed-off-by: Damian Eppel <deppel@marvell.com>
Change-Id: I3632627841b7c958a60ab3b65e314949572259c3
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/69934
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/phy/mdio-thunder.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/phy/mdio-thunder.c b/drivers/net/phy/mdio-thunder.c
index b6128ae7f14f..3d7ebe74693f 100644
--- a/drivers/net/phy/mdio-thunder.c
+++ b/drivers/net/phy/mdio-thunder.c
@@ -59,6 +59,7 @@ static int thunder_mdiobus_pci_probe(struct pci_dev *pdev,
 		struct mii_bus *mii_bus;
 		struct cavium_mdiobus *bus;
 		union cvmx_smix_en smi_en;
+		union cvmx_smix_clk smi_clk;
 
 		/* If it is not an OF node we cannot handle it yet, so
 		 * exit the loop.
@@ -87,6 +88,10 @@ static int thunder_mdiobus_pci_probe(struct pci_dev *pdev,
 		bus->register_base = (u64)nexus->bar0 +
 			r.start - pci_resource_start(pdev, 0);
 
+		smi_clk.u64 = oct_mdio_readq(bus->register_base + SMI_CLK);
+		smi_clk.s.clk_idle = 1;
+		oct_mdio_writeq(smi_clk.u64, bus->register_base + SMI_CLK);
+
 		smi_en.u64 = 0;
 		smi_en.s.en = 1;
 		oct_mdio_writeq(smi_en.u64, bus->register_base + SMI_EN);
-- 
2.34.1

